pkg/queue/breaker.go

```diff
@@ -67,6 +67,15 @@ func NewBreaker(params BreakerParams) *Breaker {
 	}
 }
 
+// Returns if the breaker can accept a request.
+func (b *Breaker) HasCapacity() bool {
+	return b.sem.hasCapacity()
+}
+
+func (s *semaphore) hasCapacity() bool {
```

Comment:
nit: Move this to the implementations of the semaphore please. Having it in between the Breaker impl. is a bit awkward.

Reply:
Done.